### PR TITLE
Add filters support to generic forwarding

### PR DESCRIPTION
**Pull request by Blinkuu**
**What this PR does**:

This PR adds support for filtering spans and span events in the [Generic Forwarding](https://grafana.com/docs/tempo/latest/operations/generic_forwarding/) feature. The filtering feature is needed for `k6 x Tempo` GA.

**Checklist**
- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by zalegrala** [comment]
I'm wondering if `pkg/spanfilter` can help here.

**Comment by Blinkuu** [comment]
> I'm wondering if `pkg/spanfilter` can help here.

The `pkg/spanfilter` does not allow filtering on `trace_id` string, which we require.

File: docs/sources/tempo/configuration/_index.md [hunk 1]
```diff
@@ -173,6 +173,15 @@ distributor:
             # Optional.
             # Path to the TLS certificate. This field must be set if insecure = false.
             [cert_file: <string | default = "">]
+       
+        # Optional.
+        # Configures filtering in forwarder that allows dropping spans and span events using
```
Comment (knylander-grafana) [inline docs/sources/tempo/configuration/_index.md:178]:
```suggestion
        # Configures filtering in forwarder that lets you drop spans and span events using
```

File: docs/sources/tempo/operations/generic_forwarding.md [hunk 2]
```diff
@@ -7,9 +7,7 @@ weight: 130
 
 # Generic forwarding
 
-Generic forwarding allows asynchronous replication of ingested traces.
-The distributor writes received spans to both the ingester and defined endpoints, if enabled.
-This feature works in a "best-effort" manner, meaning that no retries happen if an error occurs during replication.
+Generic forwarding allows asynchronous replication of ingested traces. The distributor writes received spans to both the ingester and defined endpoints, if enabled. This feature works in a "best-effort" manner, meaning that no retries happen if an error occurs during replication.
 
 >**Warning:** Generic forwarding does not work retroactively. Once enabled, the distributor only replicates freshly ingested spans.
```
Comment (knylander-grafana) [inline docs/sources/tempo/operations/generic_forwarding.md:12]: We are using an [admonitions shortcode](https://grafana.com/docs/writers-toolkit/write/style-guide/style-conventions/#admonitions). I'm suggesting this change to use the shortcode. 
```suggestion
{{% admonition type="warning" %}}
Generic forwarding does not work retroactively. Once enabled, the distributor only replicates freshly ingested spans.
{{% /admonition %}}

```

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,7 @@
 * [ENHANCEMENT] Assert ingestion rate limits as early as possible [#2640](https://github.com/grafana/tempo/pull/2703) (@mghildiy)
 * [ENHANCEMENT] Add several metrics-generator fields to user-configurable overrides [#2711](https://github.com/grafana/tempo/pull/2711) (@kvrhdn)
 * [BUGFIX] Fix panic in metrics summary api [#2738](https://github.com/grafana/tempo/pull/2738) (@mdisibio)
+* [FEATURE] Add filtering support to Generic Forwarding [#2742](https://github.com/grafana/tempo/pull/2742) (@Blinkuu)
```
Comment (kvrhdn) [inline CHANGELOG.md:7]: Nit: we group entries by type, so this one should be underneath the other `[FEATURE]`'s right above `[ENHANCEMENT]`

**Review by kvrhdn — APPROVED** [review]
This looks really good! Just left a little nit about the changelog

File: CHANGELOG.md (hunk 3 above)
Comment (Blinkuu) [reply CHANGELOG.md]: Updated in [25f4ef4](https://github.com/grafana/tempo/pull/2742/commits/25f4ef4390e7c39e2de249555008ae381476d678).

**Pull request merged**